### PR TITLE
adds new edit flow to the video block

### LINKED ISSUE #14795: Expand the new replace image flow out to other blocks
#14142 added an improved flow for replacing an image: 

swap

As per @mapk's [note in the thread](https://github.com/WordPress/gutenberg/pull/14142#issuecomment-476871253), this flow should be rolled out to other areas of the UI. Here's a quick list: 

- [ ] Video
- [ ] Audio
- [ ] Embeds* (Fixing this will also fix #11439)

In the following blocks, the edit button in the toolbar opens the media modal directly, so they just need the pencil icon swapped out for the new "Replace" icon: 

- [ ] Media & Text 
- [ ] Cover
- [ ] File

---

_*In the case of the Embed blocks (and maybe Video + Audio?) showing the preview in the placeholder may be too complicated. If that's the case, it may be fine to leave that part out._

### PR DESCRIPTION
**Pull request by draganescu**
# Description

Closes #14795

This is a follow up to the image block edit flow update which ports the new flow to all the blocks which have media with an edit state.

# How has this been tested?
For now I've only tested locally.

# Types of change
New feature (non-breaking change which adds functionality)

# Screenshots

55976311-8783fc80-5c94-11e9-9262-1ac1cb55d890

### CODE REVIEW

**Review by kjellr — COMMENTED** [review]
From a design perspective, this is all set!

video

Should get a code review before merge. 👍

File: packages/block-library/src/video/edit.js [hunk 1]
```diff
@@ -148,7 +156,7 @@ class VideoEdit extends Component {
 		} = this.props;
 		const { editing } = this.state;
 		const switchToEditing = () => {
```
Comment (jorgefilipecosta) [inline packages/block-library/src/video/edit.js]: Should we rename the function as it now does not "switchToEditing" but toggles the editing?

File: packages/block-library/src/video/edit.js [hunk 2]
```diff
@@ -164,19 +172,34 @@ class VideoEdit extends Component {
 			this.setState( { src: media.url, editing: false } );
 		};
 
+		const editImageIcon = ( <SVG width={ 20 } height={ 20 } viewBox="0 0 20 20"><Rect x={ 11 } y={ 3 } width={ 7 } height={ 5 } rx={ 1 } /><Rect x={ 2 } y={ 12 } width={ 7 } height={ 5 } rx={ 1 } /><Path d="M13,12h1a3,3,0,0,1-3,3v2a5,5,0,0,0,5-5h1L15,9Z" /><Path d="M4,8H3l2,3L7,8H6A3,3,0,0,1,9,5V3A5,5,0,0,0,4,8Z" /></SVG> );
```
Comment (jorgefilipecosta) [inline packages/block-library/src/video/edit.js]: Maybe we could move this value to the top as a constant?

**Review by jorgefilipecosta — APPROVED** [review]
I left some minor comments. The code seems ready after the minor changes are addressed 👍

**Comment by draganescu** [comment]
@jorgefilipecosta I have updated the code as per review and also rebased. Let me know if I understood your comments properly :)

**Comment by mapk** [comment]
I'm not sure if this is solely related to the Video block because I haven't tested it with others yet, but there's a **bug** when transforming the Video block to a Media+Text block and then back again to a VIdeo block. The `Edit` icon changes position in the block toolbar. This shouldn't happen.

**Expected behavior:**

Icons in the toolbar should remain in the same position for each block.

video

**Comment by draganescu** [comment]
@mapk see [this comment](https://github.com/WordPress/gutenberg/pull/15518#issuecomment-497660420) on the same issue on the embed block.

I will treat it in a separate issue.

**Comment by afercia** [comment]
Related to the "swapping" icon: https://github.com/WordPress/gutenberg/issues/16048

**Comment by draganescu** [comment]
#11952 changed direction so closing this as it became irrelevant.

**Pull request closed without merging**